packages/assets/src/loader/Loader.ts

```diff
@@ -0,0 +1,189 @@
+import { makeAbsoluteUrl } from '../utils/makeAbsoluteUrl';
+import type { LoaderParser } from './plugins/LoaderParser';
+
+export interface LoadAsset<T=any>
+{
+    src: string;
+    data?: T;
+}
+
+/**
+ * The Loader is responsible for loading all assets, such as images, spritesheets, audio files, etc.
+ * It does not do anything clever with urls - it just loads stuff!
+ * Behind the scenes all things are cached using promises. This means its impossible to load an asset more than once.
+ * Through the use of LoaderParsers, the loader can understand how to load any kind of file!
+ *
+ * It is not intended that this class is created by developers - its part of the Asset class
+ * This is the second major system of PixiJS' main Assets class
+ */
+class Loader
+{
+    /** All loader plugins registered */
+    public plugins: LoaderParser[] = [];
+
+    /** Stores all data loaded by the plugins */
+    public cache: Record<string, any> = {};
+
+    /** Cache loading promises that ae currently active */
+    public promiseCache: Record<string, Promise<any>> = {};
+
+    /** function used for testing */
+    public reset(): void
+    {
+        this.promiseCache = {};
+        this.cache = {};
+
+        this.plugins.length = 0;
+    }
+
+    /**
+     * Use this to add any plugins to the loadAssets function to use
+     *
+     * @param newParsers - a array of parsers to add to the loader, or just a single one
+     */
+    public addParser(...newParsers: LoaderParser[]): void
+    {
+        this.plugins.push(...newParsers);
+    }
```

Comment:
Suggestion: standardize on either "plugins" or "parsers", instead of mixing terms.

Reply:
thanks @andrewstart ! yeah, that some legacy bits there! going with parsers!